pipeline/pipeline/task.py

```diff
@@ -111,18 +284,105 @@ def handler(match_obj):
         return self
 
     def label(self, label):
+        """
+        Set the task's label.
+
+        Examples
+        --------
+
+        Set the task's label to `qc`:
+
+        >>> t1 = p.new_task()
+        >>> (t1.label('qc')
+        ...    .command(f'echo "hello"'))
+
+        Parameters
+        ----------
+        label: :obj:`str`
+
+        Returns
+        -------
+        :class:`.Task`
+            Same task object with label set.
+        """
         self._label = label
         return self
 
     def memory(self, memory):
+        """
+        Set the task's memory requirements.
+
+        Examples
+        --------
+
+        Set the task's memory requirement to 5GB:
+
+        >>> t1 = p.new_task()
+        >>> (t1.memory(5)
+        ...    .command(f'echo "hello"'))
+
+        Parameters
+        ----------
+        memory: :obj:`str` or :obj:`float` or :obj:`int`
+
+        Returns
+        -------
+        :class:`.Task`
+            Same task object with memory requirements set.
+        """
         self._memory = memory
         return self
 
     def cpu(self, cpu):
+        """
+        Set the task's CPU requirements.
+
+        Examples
+        --------
+
+        Set the task's CPU requirement to 2 cores:
+
+        >>> t1 = p.new_task()
+        >>> (t1.cpu(2)
+        ...    .command(f'echo "hello"'))
+
+        Parameters
+        ----------
+        cpu: :obj:`str` or :obj:`float` or :obj:`int`
```

Comment:
Cores

Reply:
Oh I really just meant in the docstring, although probably the change you made is a bit clearer as well so that's good